arangodb-net-standard/DocumentApi/DocumentApiClient.cs

```diff
@@ -286,5 +287,43 @@ public async Task<DeleteDocumentsResponse<T>> DeleteDocumentsAsync<T>(string col
             }
         }
 
+
+
+        /// <summary>
+        /// Get headers based on collection and _key with option headers.
+        /// 200: is returned if the document was found. 
+        /// 304: is returned if the “If-None-Match” header is given and the document has the same version. 
+        /// 404: is returned if the document or collection was not found. 
+        /// 412: is returned if an “If-Match” header is given and the found document has a different version. The response will also contain the found document’s current revision in the Etag header.
+        /// HEAD /_api/document/{document-handle}
+        /// </summary>
+        /// <param name="collectionName"></param>
+        /// <param name="documentKey"></param>
+        /// <param name="headers"></param>
+        /// <returns></returns>
+        public async Task<DocumentHeaderResponse> HeadDocumentAsync(string collectionName, string documentKey, HeadDocumentHeader headers = null)
+        {
+            return await HeadDocumentAsync($"{WebUtility.UrlEncode(collectionName)}/{WebUtility.UrlEncode(documentKey)}", headers);
+        }
+
+
+        /// <summary>
+        /// Get headers based on Document _id
+        /// </summary>
+        /// <param name="documentId"></param>
+        /// <param name="headers">Object containing a dictionary of Header keys and values</param>
+        /// <returns></returns>
+        public async Task<DocumentHeaderResponse> HeadDocumentAsync(string documentId, HeadDocumentHeader headers = null)
```

Comment:
Given that `HeadDocumentHeader` only contains a dictionary property for the headers key/value, should this parameter just be of type `HttpRequestHeaders`?

Otherwise I suggest we expose `If-None-Match` and `If-Match` as properties. According to the documentation, they are only used to provide "exactly one Etag". I think that would be in line with the kind of abstraction we've done up to now.

@rossmills99 Any thoughts?

Reply:
I looked at the documentation for [If-None-Match](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/If-None-Match).

So here are 2 approaches we can take:

```c#
public class HeadDocumentHeader
{
    public string IfNoneMatch { get; set; }

    public string IfMatch { get; set; }
}

public class HeadDocumentResponse
{
    public HttpStatusCode Code { get; set; }

    public string Etag { get; set; }
}
```

Or expose `HttpRequestHeaders` and `HttpResponseHeaders`.